### ISSUE TITLE
Document our standards for writing design issues in GitHub

### ISSUE THREAD

**Issue by melchoyce** [18F/delivery-assurance#89]
### A description of the work
At 18F, we tend to use GitHub issues and project boards for both engineering and design tasks. Engineering templates are fairly standardized, but design templates vary across projects, and it's often more difficult and time consuming to write a good design issue. We should take a look at writing up documentation around creating design issues in GitHub.

### Point of contact on this issue
The Design chapter ([#18f-design](https://gsa-tts.slack.com/archives/C03L05B9BC3))

**Billable?**

- [ ] Yes
- [x] No

**Skills needed**

- [x] Design
- [x] Content

**Timeline**
Does this need to happen in the next two weeks?

- [ ] Yes
- [x] No

How much time do you anticipate this work taking?

2-4 hours of drafting, another ~2 hours of review and revision.

### Acceptance Criteria

- [ ] What does "done" look like?

As a person working on an 18F project—whether I'm a product manager, engineer, member of the design chapter, or other role, I know how to properly write a design issue that makes the problem to be solved clear, and provide actionable next steps for the designer who works on said issue.

**Comment by melchoyce** [comment]
Started a brain dump here: https://docs.google.com/document/d/1di0mmGivLgN1JoMinp2rMq0ylAD3-5WU0M7ZRnczdkk/edit?usp=sharing

**Comment by austinhernandez** [comment]
I do feel like the flaw in aligning ticket structures is that it is not a 1:1 comparison. "Design" tickets could be discipline specific (content, research, visual design, product, etc.) and within these different tickets, there are different phases of work with different outputs. There is then the complications of team structure (e.g. working with just designers or a "traditional" product team) and how that team is working with design.

Looking at your brain dump and comments, I see two potential solutions:

1. Multiple ticket-templates that cover a few of different use cases, potential outputs, etc.
2. One ticket template that has a very broad "who, what, why" format that can be applied across design disciplines, deliverables, team structures, etc.

I see the potential pitfalls being either too prescriptive or too broad, respectively. I am leaning on too broad as it is our job as designers (of all disciplines) to answer these "who, what, why" questions. The ticket (especially the acceptance criteria) should contain the how.

Here's a proposal for a general design ticket. Let's try to break this format—why wouldn't something like this work? What would make it work? 

> 
> **User story format**
> As <who>, I need <what> so that <why>.
> 
> **Background**
> Previous context people need to know (related tickets, conversations, etc.)
> 
> **Questions**
> For stand-ups, future designers, peer reviewers, for product etc.
> 
> **Acceptance Criteria**
> How do we resolve the who, what, and why of this ticket?
> 
> **Notes**


PS I can add comments and thoughts in your [Document our standards for writing design issues in GitHub](https://docs.google.com/document/d/1di0mmGivLgN1JoMinp2rMq0ylAD3-5WU0M7ZRnczdkk/edit) doc, too

**Comment by melchoyce** [comment]
Going to take this format for a spin, based on some recent DA design issues. Here's the first.

## Update 18F icon packs #81

> ### User story format
> As an 18F team member who uses our icon set regularly to create clear, effective communications with partners, executive sponsors, and internal teams, I need our icons to stay current with new iconographic standards and look polished, so that they look professional and help me prove my expertise.
>  
> ### Background
> 
> Our icons have fallen a little out of date, and could use some maintenance:
> 
> 1. **Review and fix end caps**
> Here's an example on the `zoom-out` icon export: where the handle meets the glass, the end cap should not be visible; but instead it is rounded and sticks out.
> ![zoom-out](https://user-images.githubusercontent.com/2374206/224725454-02d1ee51-3cd0-4a36-ae5a-158807087824.png)
> 2. Add 18F dark blue export to downloads (#1C304A)
> 3. Update [downloads](https://brand.18f.gov/icons/) on 18F identity site (sketch files, exports, sticker sheets), [slide starters](https://docs.google.com/presentation/d/11E9IBb1JkNWz5FLkipXJCmhtKJQa2EL3h0W3EqPZYNQ/edit#slide=id.g5ecdbbee9e_0_0), and [18F deck template](https://brand.18f.gov/templates/) as needed
> 
> #### References
> - [Icon sets](https://brand.18f.gov/icons/)
> - [Slide starter packs](https://docs.google.com/presentation/d/11E9IBb1JkNWz5FLkipXJCmhtKJQa2EL3h0W3EqPZYNQ/edit#slide=id.g5ecdbbee9e_0_0)
> 
> ### Questions
> - What’s people’s experience using these shared assets in presentations?
> - What do people need that we don’t have?
> - What visualizations are missing?
> 
> ### Acceptance Criteria
> 
> 1. [ ] All icons with protruding end caps have been fixed
> 2. [ ] Dark blue icons have been exported
> 3. [ ] All updated icons have been added to [downloads](https://brand.18f.gov/icons/) on 18F identity site (sketch files, exports, sticker sheets) and [slide starters](https://docs.google.com/presentation/d/11E9IBb1JkNWz5FLkipXJCmhtKJQa2EL3h0W3EqPZYNQ/edit#slide=id.g5ecdbbee9e_0_0)
> 
> ### Notes
> 
> Note additional issues as needed. The team should feel empowered to reorganize and adjust assets so that they work better.

This issue fits the format pretty well, though I'm noticing we'll want a point-of-contact field for follow-up questions/to identify any additional stakeholders or SMEs.

**Comment by melchoyce** [comment]
Trying out something like https://github.com/18F/tock/issues/1565, which straddles the line between design and dev:

> ### User story format
> As an 18F member with low vision, I need Tock to pass AA color contrast so that I can easily report my time.
> 
> ### Background
> 
> #### Steps to Reproduce:
> Navigate to https://tock.18f.gov/reporting_period/#
> Scroll down to 'add item', "billable hours" link
> 
> The blue link does not meet color contrast against the blue background (WCAG Success Criteria (A): 1.4.3 Contrast). We should update either the background or link color so there is adequate contrast.
> 
> ### Acceptance Criteria
> [ ] The colors have been updated so they pass color contrast requirements
> [ ] The update has been pushed live to Tock

This one doesn't work as well, and would also benefit from having a section just for adding screenshots IMO.

**Comment by melchoyce** [comment]
One last trial:

> ## Create a checklist/guidelines for accessible deliverables #87
> 
> ### User story format
> - As an 18F staff member who is preparing a new slide deck, I want to find out what actions to take to ensure both the file and the presentation itself will be accessible.
> - As an 18F staff member who is creating a new document (text, table, chart, or diagram), I want to find out what actions to take to ensure it is accessible.
> 
> ### Background
> 
> Create a checklist or guidelines for ensuring accessible "everyday" deliverables — slides/presentation tips, documents, tables, charts, and diagrams. This work would help with onboarding, consistency, and practicing accessibility every day. It would also help us model our commitment to accessibility outside of software design/development.
> 
> #### Why now?
> 
> 1. Building on recent training: Many team members just completed WebAIM accessibility training, and can apply the information learned during this outside of web contexts.
> 2. Creating a solid foundation: As people use delivery assurance time to develop internal training and materials, this is a way to ensure that work is accessible from the start.
> 
> ### Acceptance Criteria
> - [ ] As an 18F staff member, I can locate this checklist
> - [ ] As an 18F staff member, I understand how to use this checklist to ensure my work is accessible
> 
> ### Notes
> I'd also want this to be part of the resource library and shared in #g-accessibility.

This partly works, but I've dropped the ask into the Background section which doesn't quite work, and might actually obscure the task being requested.

**Comment by katefisher808** [comment]
@melchoyce love this!! the only two questions I might have as a designer/researcher taking on an issue in this format would be: 1) what work has been done already? or what work does this build off of? and 2) who are the ppl I should talk to about the issue and why. I could see the template accounts for those things in the 'background' section

**Comment by melchoyce** [comment]
Does this feel like too much?

> ## User story
> As `<who>`, I need `<what>` so that `<why>`.
>
> ### Additional information about the request
> 
> ## Background
> Previous context people need to know (related tickets, conversations, etc.), including prior work.
> 
> ### Open Questions
> For stand-ups, future designers, peer reviewers, for product etc.
> 
> ### Point of contact on this issue
> Who can we follow-up with if we have questions?
> 
> ## Additional Details
> ### Billable?
> 
> - [ ] Yes
> - [ ] No
> If yes, tock code:
> 
> ### Timeline
> 
> Does this need to happen in the next two weeks?
> - [ ] Yes
> - [ ] No
> 
> ### How much time do you anticipate this work taking?
> Hours, weeks, or months?
> 
> ## Acceptance Criteria
> How do we resolve the who, what, and why of this ticket?

**Comment by katefisher808** [comment]
I think it works well!

**Comment by austinhernandez** [comment]
My thoughts: 

- I think it's a good first step!
- Making the ticket shouldn't take longer than the task
  - (That's why I am all for larger ticket with smaller tasks as acceptance criteria.)
- In my experience, someone always got strong opinions on the "correct" way to make a ticket.
   - Maybe we should pilot this ticket on some DA work, see how its going, then work from there.

**Comment by melchoyce** [comment]
Thanks all. I've created a PR for a new issue template that we can test out. 

Next steps:

1. I'll open a new issue once the PR is merged to evaluate whether or not it's working for us.
2. There's still the question of if we want to provide 18F-wide documentation around writing design issues on GitHub. [I have a draft doc started](https://docs.google.com/document/d/1di0mmGivLgN1JoMinp2rMq0ylAD3-5WU0M7ZRnczdkk/edit) which could use more review.
3. If we do want to create documentation, where should it live? How can we ensure people find it?

**Comment by austinhernandez** [comment]
I have no idea off the top of my head where the documentation would live. Where does the engineering one live? Is it part of onboarding? Is it role specific? 

I know that documentation is in the scope of this ticket, but I might create a follow up ticket for #130 for documentation _after_ feedback. You can use the draft you started here, but having more feedback might help inform where documentation would live. I feel like either way, the way you write the documentation would have to be informed by how people are using it.
